src/Spice86.Core/Emulator/Devices/Sound/Midi/MT32/Mt32MidiDevice.cs

```diff
@@ -21,32 +28,116 @@ internal sealed class Mt32MidiDevice : MidiDevice {
     /// Constructs an instance of <see cref="Mt32MidiDevice"/>.
     /// </summary>
-    /// <param name="softwareMixer">The emulator's software mixer for all sound channels.</param>
+    /// <param name="softwareMixer">The software mixer for sund channels.</param>
     /// <param name="romsPath">The path to the MT-32 ROM files.</param>
     /// <param name="loggerService">The logger service to use for logging messages.</param>
-    /// <param name="pauseHandler">The handler for pausing the emulator</param>
     /// <exception cref="ArgumentNullException">Thrown if <paramref name="romsPath"/> is <c>null</c> or empty.</exception>
-    public Mt32MidiDevice(SoftwareMixer softwareMixer, string romsPath, ILoggerService loggerService, IPauseHandler pauseHandler) {
+    public Mt32MidiDevice(SoftwareMixer softwareMixer, string romsPath, ILoggerService loggerService) {
+        _soundChannel = softwareMixer.CreateChannel(nameof(Mt32MidiDevice));
+        _context = new();
         if (string.IsNullOrWhiteSpace(romsPath)) {
             throw new ArgumentNullException(nameof(romsPath));
         }
-        _player = new Mt32Player(softwareMixer, romsPath, loggerService, pauseHandler);
+        
+        if (!LoadRoms(romsPath)) {
+            if (loggerService.IsEnabled(Serilog.Events.LogEventLevel.Error)) {
+                loggerService.Error("{MethodName} could not find roms in {RomsPath}, {ClassName} was not created",
+                    nameof(LoadRoms), romsPath, nameof(Mt32MidiDevice));
+            }
+            return;
+        }
+        _context.AnalogOutputMode = Mt32GlobalState.GetBestAnalogOutputMode(48000);
+        _context.SetSampleRate(48000);
+        _context.OpenSynth();
+
+        _renderThread = new Thread(RenderThreadMethod) {
+            Name = nameof(Mt32MidiDevice)
+        };
     }
 
     /// <inheritdoc/>
-    protected override void PlayShortMessage(uint message) => _player.PlayShortMessage(message);
+    protected override void PlayShortMessage(uint message) {
+        StartThreadIfNeeded();
+        if (!_disposed && !_exitRenderThread) {
+            _context.PlayMessage(message);
+            RaiseFillBufferEvent();
+        }
+    }
 
     /// <inheritdoc/>
-    protected override void PlaySysex(ReadOnlySpan<byte> data) => _player.PlaySysex(data);
+    protected override void PlaySysex(ReadOnlySpan<byte> data) {
+        StartThreadIfNeeded();
+        if (!_disposed && !_exitRenderThread) {
+            _context.PlaySysex(data);
+            RaiseFillBufferEvent();
+        }
+    }
+
+
+    private void StartThreadIfNeeded() {
+        if (!_disposed && !_exitRenderThread && !_threadStarted) {
+            _threadStarted = true;
+            _renderThread?.Start();
+        }
+    }
+
+    private void RenderThreadMethod() {
+        Span<float> buffer = stackalloc float[128];
+        while (!_exitRenderThread) {
+            if (!_exitRenderThread) {
+                _fillBufferEvent.WaitOne(1);
```

Comment:
The _fillBufferEvent is never reset so this wil only wait 1ms one time and then never again.

If thread sysnchronization is the issue this tries to solve then we established that a `Thread.Sleep(0)` is a lot easier.

Reply:
This has been removed.